File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/ExperienceRequestDto.java
package com.ungs.docsys.dtos;

import java.time.LocalDate;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ExperienceRequestDto {
    @NotBlank(message = "Job title must not be blank")
    @Size(max = 150, message = "Job title must not exceed 150 characters")
    private String jobTitle;
    @NotBlank(message = "Company name must not be blank")
    @Size(max = 150, message = "Company name must not exceed 150 characters")
    private String companyName;
    @NotBlank(message = "Description must not be blank")
    @Size(max = 5000, message = "Description must not exceed 5000 characters")
    private String description;
    @NotNull(message = "Start date must not be null")
    private LocalDate startDate;
    private LocalDate endDate;
    @NotNull(message = "isCurrentJob must not be null")
    private Boolean isCurrentJob;
}

File: ungs-docsys-bff/src/modules/resume-user/dtos/experience-request.dto.ts
import {
  IsString,
  MaxLength,
  IsDateString,
  IsOptional,
  IsBoolean
} from 'class-validator';

export class ExperienceRequestDto {
  @IsOptional()
  @IsString()
  @MaxLength(150, { message: 'Job title must not exceed 150 characters' })
  jobTitle?: string;

  @IsOptional()
  @IsString()
  @MaxLength(150, { message: 'Company name must not exceed 150 characters' })
  companyName?: string;

  @IsOptional()
  @IsString()
  @MaxLength(5000, { message: 'Description must not exceed 5000 characters' })
  description?: string;

  @IsOptional()
  @IsDateString({}, { message: 'Start date must be a valid ISO 8601 date string' })
  startDate?: string;

  @IsOptional()
  @IsDateString({}, { message: 'End date must be a valid ISO 8601 date string' })
  endDate?: string;

  @IsOptional()
  @IsBoolean({ message: 'isCurrentJob must be a boolean value' })
  isCurrentJob?: boolean;
}